### PR TITLE
PCI-251 Improve purge-chain command with clear also rocks_db_storage.

### PR DESCRIPTION
**Pull request by maryjanyes**
Add a new command for node_template `purge-abci-storage` to purge abci_storage_rocksdb directory;
Rename `on_new_session` method name to `call_on_new_session`.

### CODE REVIEW

File: src/command.rs [hunk 1]
```diff
@@ -118,6 +118,10 @@ pub fn run() -> sc_cli::Result<()> {
             let runner = cli.create_runner(cmd)?;
             runner.sync_run(|config| cmd.run(config.chain_spec, config.network))
         }
+        Some(Subcommand::PurgeAbciStorage(ref cmd)) => {
```
Comment (Mr-Leshiy) [inline src/command.rs:121]: We do not need to have to different cmd commands PurgeAbciStorage and PurgeChain. It should be only one cmd command PurgeChain which will remove all storages: original and our storage.

File: src/cli/commands.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#[derive(Debug, StructOpt)]
+pub struct PurgeAbciStorageCmd {
+    #[allow(missing_docs)]
+    #[structopt(flatten)]
+    pub shared_params: sc_cli::SharedParams,
+}
+
+impl PurgeAbciStorageCmd {
+    /// Run the purge command.
+    pub fn run(&self, config: &Configuration) -> sc_cli::Result<()> {
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:20]: This function should processed the ```-y``` flag during the invoking it. Look at the PureChainCmd srtuct.

**Review by Mr-Leshiy — CHANGES_REQUESTED** [review]

**Comment by Mr-Leshiy** [comment]
Remove this function ```remove_abci_storage()``` from the command.rs

File: src/cli/commands.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+use sc_service::Configuration;
+use std::fmt::Debug;
+use std::{fs, io, path};
+use structopt::StructOpt;
+
+pub use pallet_abci;
+
+/// The `purge-abci-storage` command used to remove abci storage.
+#[derive(Debug, StructOpt)]
+pub struct PurgeChainWithStorageCmd {
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:11]: add ```DataBaseParams``` as a field of this struct

File: src/cli/commands.rs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            },
+        };
+
+        let _o = purge_chain_cmd.run(config.database.clone());
+
+        Ok(())
+    }
+}
+
+impl CliConfiguration for PurgeChainWithStorageCmd {
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:90]: add ```fn database_params(&self) -> Option<&DatabaseParams>``` method like in the PurgeChainCmd

File: src/cli/commands.rs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                    println!("Aborted");
+                }
+            }
+        } else {
+            remove_rocks_db(rocks_db_name, db_path);
+        }
+
+        let purge_chain_cmd = sc_cli::PurgeChainCmd {
+            yes: false,
+            shared_params: sc_cli::SharedParams {
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:72]: Copy ```shared_params```, do not construct it using constructor

File: src/cli/commands.rs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                dev: shared_params.dev,
+                chain: None,
+                base_path: None,
+                log: shared_params.log.clone(),
+            },
+            database_params: sc_cli::DatabaseParams {
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:78]: same

File: src/cli/commands.rs [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+}
+
+impl PurgeChainWithStorageCmd {
+    /// Run the purge command.
+    pub fn run(&self, config: &Configuration) -> sc_cli::Result<()> {
+        let chain_spec_id = config.chain_spec.id();
+        let shared_params = self.shared_params();
+        let database_params = self.database_params().unwrap_or(&sc_cli::DatabaseParams {
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:35]: remove this

File: src/cli/commands.rs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            println!("{:?} did not exist.", &db_path);
+        }
+    }
+}
+
+impl PurgeChainWithStorageCmd {
+    /// Run the purge command.
+    pub fn run(&self, config: &Configuration) -> sc_cli::Result<()> {
+        let chain_spec_id = config.chain_spec.id();
+        let shared_params = self.shared_params();
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:34]: remove this

File: src/cli/commands.rs [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+            .unwrap()
+            .path()
+            .to_path_buf()
+            .join("chains")
+            .join(chain_spec_id);
+        let rocks_db_name = &pallet_abci::get_storage_name();
+        let db_path = config_dir.join(&rocks_db_name);
+
+        if !self.yes {
+            println!("Are you sure to remove {:?}? [y/N]: ", &db_path);
```
Comment (Mr-Leshiy) [inline src/cli/commands.rs:52]: Ask that you want to remove all databases not only rocks db. If answer is 'yes' remove both database, the same with the non-interactive mode. 
As a result application should not ask user TWICE to reset the state.

**Review by Mr-Leshiy — CHANGES_REQUESTED** [review]

**Review by enfipy — APPROVED** [review]

**Review by Mr-Leshiy — APPROVED** [review]

**Pull request merged**